chart/epinio/templates/stage-scripts.yaml

```diff
@@ -62,8 +66,14 @@ data:
     # container gracefully. We use `|| true` in case linkerd is not deployed.  Further, it
     # is placed into a trap to ensure that it will always run, even for a staging failure.
     #
+    set -e
     trap "curl -X POST http://localhost:4191/shutdown || true" EXIT
     echo By _ _ __ ___ _____ $(whoami) $(pwd)
+    if test ! -d "/workspace/source/app" ; then
+      echo Nothing to build
+      sleep 60 # linkerd is a pain - If we exit to quickly, with the sidecar not ready our curl to shut it down does nothing, and then the sidecar comes up and prevents the pod from ending
```

Comment:
The only reason we still keep the staging Job meshed is because it may talk to minio to fetch the source blob (if external S3 is not used).

The helm chart [we are planning to use](https://github.com/epinio/helm-charts/pull/96) supports passing a secret with certs which should make it possible to use cert-manager to create a cert and encrypt communication with minio: https://github.com/minio/minio/blob/master/helm/minio/values.yaml#L110-L116

If that works, we can "unmesh" the staging job and get rid of this trick.

Reply:
Yes, having this in the future is pretty much the only reason I am ok with my hack here.
Not as a hacky, more complicated would have been code waiting for the sidecar to be up ...
Basically repeat the `/shutdown` until it connects and succeeds.